sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.aest.controller.getApplicationVersionHistory.e2e-spec.ts

```diff
@@ -0,0 +1,171 @@
+  getAESTToken,
+} from "../../../testHelpers";
+import {
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import { ApplicationStatus } from "@sims/sims-db";
+import { addDays, getPSTPDTDateHourMinute } from "@sims/utilities";
+
+describe("ApplicationAESTController(e2e)-getApplicationVersionHistory", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should get an array of application versions when there are an overwritten application and a completed application associated with the given application.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(db.dataSource, undefined, {
+      submittedDate: new Date(),
+    });
+
+    const overwrittenApplication = await saveFakeApplication(
+      db.dataSource,
+      {
+        institution:
+          application.currentAssessment.offering.institutionLocation
+            .institution,
+        institutionLocation:
+          application.currentAssessment.offering.institutionLocation,
+        student: application.student,
+        program: application.currentAssessment.offering.educationProgram,
+        offering: application.currentAssessment.offering,
+        programYear: application.programYear,
+      },
+      {
+        applicationStatus: ApplicationStatus.Overwritten,
+        applicationNumber: application.applicationNumber,
+        offeringIntensity:
+          application.currentAssessment.offering.offeringIntensity,
+        pirStatus: application.pirStatus,
+        submittedDate: addDays(-10),
+      },
+    );
+
+    const completedApplication = await saveFakeApplication(
+      db.dataSource,
+      {
+        institution:
+          application.currentAssessment.offering.institutionLocation
+            .institution,
+        institutionLocation:
+          application.currentAssessment.offering.institutionLocation,
+        student: application.student,
+        program: application.currentAssessment.offering.educationProgram,
+        offering: application.currentAssessment.offering,
+        programYear: application.programYear,
+      },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+        applicationNumber: application.applicationNumber,
+        offeringIntensity:
+          application.currentAssessment.offering.offeringIntensity,
+        pirStatus: application.pirStatus,
+        submittedDate: addDays(-5),
+      },
+    );
+
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    const endpoint = `/aest/application/${application.id}/versions`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
```

Comment:
We can expect response like this. 

![image](https://github.com/user-attachments/assets/3dd1a92c-c8f8-4524-9c17-ff9c90712d7f)